scipy/signal/spectral.py

```diff
@@ -267,13 +267,179 @@ def welch(x, fs=1.0, window='hanning', nperseg=256, noverlap=None, nfft=None,
     2.0077340678640727
 
     """
-    x = np.asarray(x)
+    [f, Pxx] = csd(x, x, fs, window, nperseg, noverlap, nfft, detrend, 
+                   return_onesided, scaling, axis)
+
+    return f, Pxx
+
+def coherence(x, y, fs=1.0, window='hanning', nperseg=256, noverlap=None,
+              nfft=None, detrend='constant', axis=-1):
+    """
+    Estimate the magnitude squared coherence estimate, Cxy, of discrete-time 
+    signals X and Y Welch's method. Cxy = abs(Pxy)**2/(Pxx*Pyy), where Pxx and 
+    Pyy are power spectral density estimates of X and Y, and Pxy is the cross
+    spectral density estimate of X and Y. 
+
+    Parameters
+    ----------
+    x : array_like
+        Time series of measurement values
+    y : array_like
+        Time series of measurement values
+    fs : float, optional
+        Sampling frequency of the `x` and `y` time series in units of Hz. 
+        Defaults to 1.0.
+    window : str or tuple or array_like, optional
+        Desired window to use. See `get_window` for a list of windows and
+        required parameters. If `window` is array_like it will be used
+        directly as the window and its length will be used for nperseg.
+        Defaults to 'hanning'.
+    nperseg : int, optional
+        Length of each segment.  Defaults to 256.
+    noverlap: int, optional
+        Number of points to overlap between segments. If None,
+        ``noverlap = nperseg / 2``.  Defaults to None.
+    nfft : int, optional
+        Length of the FFT used, if a zero padded FFT is desired.  If None,
+        the FFT length is `nperseg`. Defaults to None.
+    detrend : str or function or False, optional
+        Specifies how to detrend each segment. If `detrend` is a string,
+        it is passed as the ``type`` argument to `detrend`.  If it is a
+        function, it takes a segment and returns a detrended segment.
+        If `detrend` is False, no detrending is done.  Defaults to 'constant'.
+    axis : int, optional
+        Axis along which the CSD is computed for both inputs; the default is 
+        over the last axis (i.e. ``axis=-1``).
+
+    Returns
+    -------
+    f : ndarray
+        Array of sample frequencies.
+    Cxy : ndarray
+        Magnitude squared coherence of x and y.
+
+    See Also
+    --------
+    periodogram: Simple, optionally modified periodogram
+    lombscargle: Lomb-Scargle periodogram for unevenly sampled data
+    welch: Power spectral density by Welch's method.
+    csd: Cross spectral density by Welch's method. 
+
+    Notes
+    --------
+    An appropriate amount of overlap will depend on the choice of window
+    and on your requirements.  For the default 'hanning' window an
+    overlap of 50% is a reasonable trade off between accurately estimating
```

Comment:
Either this needs to be ``\%`` or the whole docstring needs to be a raw string (start with ``r"""``). Otherwise ``%`` is interpreted as a special symbol.